packages/flutter/lib/src/rendering/slivers.dart

```diff
@@ -0,0 +1,1850 @@
+      if (childLayoutGeometry.opaqueExtent > 0.0)
+        minPaintOffset = layoutOffset + childLayoutGeometry.opaqueExtent;
+      scrollOffset -= childLayoutGeometry.scrollExtent;
+      layoutOffset += childLayoutGeometry.layoutExtent;
+
+      if (scrollOffset <= 0.0) {
+        minPaintOffset = layoutOffset;
+        scrollOffset = 0.0;
+      } else {
+        minPaintOffset = math.max(minPaintOffset, layoutOffset + scrollOffset);
+      }
+
+      // out-of-band data mutation
+      switch (growthDirection) {
+        case GrowthDirection.forward:
+          _maxScrollExtent += childLayoutGeometry.scrollExtent;
+          break;
+        case GrowthDirection.reverse:
+          _minScrollExtent -= childLayoutGeometry.scrollExtent;
+          break;
+      }
+      _shrinkWrapExtent += childLayoutGeometry.maxPaintExtent;
+
+      // move on to the next child
+      assert(child.parentData == childParentData);
+      child = advance(child);
+    }
+
+    // we made it without a correction, whee!
+    return 0.0;
+  }
+
+  Offset _computeAbsolutePaintOffset(RenderSliver child, double paintOffset, GrowthDirection growthDirection) {
+    assert(axisDirection != null);
+    assert(growthDirection != null);
+    switch (applyGrowthDirectionToAxisDirection(axisDirection, growthDirection)) {
+      case AxisDirection.up:
+        return new Offset(0.0, size.height - (paintOffset + child.geometry.paintExtent));
+      case AxisDirection.right:
+        return new Offset(paintOffset, 0.0);
+      case AxisDirection.down:
+        return new Offset(0.0, paintOffset);
+      case AxisDirection.left:
+        return new Offset(size.width - (paintOffset + child.geometry.paintExtent), 0.0);
+    }
+    return null;
+  }
+
+  @override
+  void applyPaintTransform(RenderObject child, Matrix4 transform) {
+    assert(child != null);
+    assert(child.parent == this);
+    final SliverPhysicalParentData childParentData = child.parentData;
+    childParentData.applyPaintTransform(transform);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (center == null) {
+      assert(firstChild == null);
+      return;
+    }
+    assert(center.parent == this);
+    assert(firstChild != null);
+    RenderSliver child;
+    // TODO(ianh): if we have content beyond our max extents, clip
+    child = firstChild;
+    while (child != center) {
+      if (child.geometry.visible) {
+        final SliverPhysicalParentData childParentData = child.parentData;
+        context.paintChild(child, offset + childParentData.paintOffset);
+      }
+      child = childAfter(child);
+    }
+    child = lastChild;
+    while (true) {
+      if (child.geometry.visible) {
+        final SliverPhysicalParentData childParentData = child.parentData;
+        context.paintChild(child, offset + childParentData.paintOffset);
+      }
+      if (child == center)
+        break;
+      child = childBefore(child);
```

Comment:
Couldn't you, just as easily, want an app-bar like effect at the bottom of the screen?  It just seems like an odd sop to headers over footers given that we paint almost everything else in child-order.